src/gui/PointViewerMainWindow.cpp

```diff
@@ -397,6 +397,18 @@ void PointViewerMainWindow::handleMessage(QByteArray message)
         }
         m_geometries->unloadFiles(regex);
     }
+    else if (commandTokens[0] == "CENTER_ON_LABEL")
+    {
+        QString regex_str = commandTokens[1];
+        QRegExp regex(regex_str, Qt::CaseSensitive, QRegExp::FixedString);
+        if (!regex.isValid())
+        {
+            g_logger.error("Invalid pattern in -unload command: '%s': %s",
+                           regex_str, regex.errorString());
+            return;
+        }
+        m_pointView->centerOnGeometry(m_geometries->findLabel(regex));
```

Comment:
What happens if `findLabel()` returns an invalid QModelIndex (may be tested with `isValid()`)?